db/migrate/20251023121031_add_payment_method_to_core_tables.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+class AddPaymentMethodToCoreTables < ActiveRecord::Migration[8.0]
+  disable_ddl_transaction!
+
+  def change
+    create_enum :payment_method_types, %w[provider manual]
+
+    safety_assured do
+      change_table :subscriptions, bulk: true do |t|
+        t.references :payment_method, type: :uuid, null: true, index: {algorithm: :concurrently}
+        t.enum :payment_method_type, enum_type: "payment_method_types", default: "provider", null: false
+      end
+    end
+    add_foreign_key :subscriptions, :payment_methods, validate: false
+
+    safety_assured do
+      change_table :wallets, bulk: true do |t|
```

Comment:
I think we should avoid migrating all table in the same migration file. Specially for wallet as this table is already known for beeing "touchy" 😅